api/src/main/java/org/jboss/forge/roaster/model/util/Types.java

```diff
@@ -30,11 +35,14 @@
    // [S=short,
    // [Z=boolean
    private static final Pattern CLASS_ARRAY_PATTERN = Pattern.compile("\\[+(B|F|C|D|I|J|S|Z|L)([0-9a-zA-Z\\.\\$]*);?");
-   private static final Pattern SIMPLE_ARRAY_PATTERN = Pattern
-            .compile("((?:[0-9a-zA-Z\\$]+)(?:\\<[^\\.^\\[]+)?(?:\\.(?:[0-9a-zA-Z\\$]+)(?:\\<[^\\.^\\[]+)?)*)(\\[\\])+");
-   private static final Pattern GENERIC_PATTERN = Pattern.compile(".*<.*>$");
+
+    private static final Pattern SIMPLE_ARRAY_PATTERN = Pattern
+            .compile("^((.)+)(\\[\\])+$");
+    private static final Pattern IDENTIFIER_PATTERN = Pattern
```

Comment:
This is excellent. I actually had no idea you could call Character methods through the POSIX \\p. Looks like I need to add something to my regular expression tutorials!